src/py/flwr/server/driver/grpc_driver.py

```diff
@@ -206,39 +213,51 @@ def push_messages(self, messages: Iterable[Message]) -> Iterable[str]:
             msg_proto = message_to_proto(msg)
             # Add to list
             message_proto_list.append(msg_proto)
-        # Call GrpcDriverStub method
-        res: PushInsMessagesResponse = self._stub.PushMessages(
-            PushInsMessagesRequest(
-                messages_list=message_proto_list, run_id=cast(Run, self._run).run_id
-            )
-        )
-        if len([msg_id for msg_id in res.message_ids if msg_id]) != len(
-            list(message_proto_list)
-        ):
-            log(
-                WARNING,
-                "Not all messages could be pushed to the SuperLink. The returned "
-                "list has `None` for those messages (the order is preserved as passed "
-                "to `push_messages`). This could be due to a malformed message.",
+
+        try:
+            # Call GrpcDriverStub method
+            res: PushInsMessagesResponse = self._stub.PushMessages(
+                PushInsMessagesRequest(
+                    messages_list=message_proto_list, run_id=cast(Run, self._run).run_id
+                )
             )
-        return list(res.message_ids)
+            if len([msg_id for msg_id in res.message_ids if msg_id]) != len(
+                list(message_proto_list)
```

Comment:
I don't think we need to convert it to list, as it's already a list, no?
```suggestion
                message_proto_list
```

Reply:
Good catch! (old code)